### PR TITLE
minio service: add additional config options

### PR DESCRIPTION
**Pull request by bachp**
###### Motivation for this change

Set access and secret key and disable browser.
Tests extended to do real operations against minio.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/web-servers/minio.nix [hunk 1]
```diff
@@ -29,6 +29,38 @@ in
       description = "The config directory, for the access keys and other settings.";
     };
 
+    accessKey = mkOption {
+      default = "";
+      type = types.string;
```
Comment (joachifm) [inline nixos/modules/services/web-servers/minio.nix]: I expect you want `types.str` (`string` has potentially surprising merge semantics & is deprecated).

Comment (bachp) [inline nixos/modules/services/web-servers/minio.nix]: Thanks for pointing this out. I wasn't even aware of the difference.

**Review by 0xABAB — APPROVED** [review]

File: nixos/modules/services/web-servers/minio.nix [hunk 2]
```diff
@@ -29,6 +29,38 @@ in
       description = "The config directory, for the access keys and other settings.";
     };
 
+    accessKey = mkOption {
+      default = "";
+      type = types.str;
+      description = ''
+        Specify the Secret key of 8 to 40 characters in length that clients use to access the server.
+        If not set the generated key from <literal>configDir</literal> is used.
```
Comment (Mic92) [inline nixos/modules/services/web-servers/minio.nix]: There should be an option to not store secrets in the nix store. For background: https://github.com/NixOS/nixpkgs/issues/24288

File: nixos/modules/services/web-servers/minio.nix [hunk 3]
```diff
@@ -29,6 +29,38 @@ in
...
+      '';
+    };
+
+    secretKey = mkOption {
+      default = "";
+      type = types.str;
+      description = ''
+        Access key of 5 to 20 characters in length that clients use to access the server.
+        If not set the generated key from <literal>configDir</literal> is used.
```
Comment (Mic92) [inline nixos/modules/services/web-servers/minio.nix]: ok. so this means, when somebody prefer not store secrets in the nix store, the can fall back to this method? (using a generated one)

Comment (bachp) [inline nixos/modules/services/web-servers/minio.nix]: Yes exactly. Do you think I need to clarify this?

Comment (0xABAB) [inline nixos/modules/services/web-servers/minio.nix]: If I just read the 4 line context visible here, I have no idea where this generated key comes from. Generated by what process and who designed that process? Upstream/you?

Comment (bachp) [inline nixos/modules/services/web-servers/minio.nix]: @Mic92 @0xABAB I tried to rephrase the description to make it more where the generated keys come from.

**Review by 0xABAB — APPROVED** [review]

**Pull request merged**